tico/experimental/quantization/passes/insert_quantize_on_dtype_mismatch.py

```diff
@@ -210,6 +212,39 @@ def _insert_quantize_op_after(node):
                     logger.debug(
                         f"quantize_per_tensor.default is inserted after {node.name}."
                     )
+                else:
+                    raise NotYetSupportedError(
+                        f"Unsupported dtype: From {qparam_dtype(inp)} to {qparam_dtype(node)}"
+                    )
+
+            elif node.target == torch.ops.aten.add.Tensor:
+                add_args = AddTensorArgs(*node.args, **node.kwargs)
+                x = add_args.input
+                y = add_args.other
+
+                if not isinstance(x, torch.fx.Node):
+                    continue
+                if not isinstance(y, torch.fx.Node):
+                    continue
+
+                if QPARAM_KEY not in x.meta:
+                    continue
+                if QPARAM_KEY not in y.meta:
+                    continue
+                if QPARAM_KEY not in node.meta:
+                    continue
+
+                if qparam_dtype(x) == qparam_dtype(node):
+                    continue
+
+                if qparam_dtype(x) == "int16" and qparam_dtype(node) == "uint8":
+                    quantize = _insert_quantize_op_after(node)
+
```

Comment:
Is it okay without checking qparam_dtype(y)? What if one of x and y get folded in another pass and dtype gets differ?

Reply:
Sounds reasonable. I will add a check for that case.